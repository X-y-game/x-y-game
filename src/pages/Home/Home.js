import React from "react";
import styled from "styled-components";
import ReactRouterPropTypes from "react-router-prop-types";
import poster from "../../assets/logo.png";
import soundPlay from "../../utils/sound";

export default function Home({ history }) {
  const handleClick = () => {
    soundPlay("click");
    history.push("/channel");
  };

  return (
    <Wrapper>
      <GameIntro>
        <div>
          <Content>
            <Poster src={poster} alt="game-poster" />
          </Content>
        </div>
        <Button type="button" onClick={handleClick}>
          Start
        </Button>
      </GameIntro>
    </Wrapper>
  );
}

Home.propTypes = {
  history: ReactRouterPropTypes.history.isRequired,
};

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  width: 100%;
  height: 100%;
`;

const GameIntro = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  h1 {
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 2rem;
    margin-bottom: 21px;
  }

  p {
    font-size: 1.5em;
    font-weight: 500;
    text-align: center;
    margin: 1.2em 0 1.5em 0;
    text-transform: uppercase;
  }

  @media screen and (max-height: 600px) {
    h1,
    p {
      font-size: 1rem;
      margin-bottom: 0;
    }
  }
`;

const Content = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  img {
    width: 30%;
    @media screen and (max-width: 1000px) {
      width: 55%;
      transition: 0.1s;
    }
    @media screen and (max-width: 600px) {
      width: 90%;
      transition: 0.1s;
    }
    @media screen and (max-height: 600px) {
      width: 30%;
      transition: 0.1s;
    }
  }
`;

const Button = styled.button`
  width: 25%;
  height: 2.5em;
  font-size: 1.2rem;
  border-radius: 1.2rem;
  background-color: #54628c;
  margin-top: 2em;
  cursor: pointer;
  color: #fff;
  outline: 0;

  &:active {
    color: #f2aeae;
  }
`;

const Poster = styled.img`
  width: 50%;
  height: 50%;
  border-radius: 1.2rem;
`;
