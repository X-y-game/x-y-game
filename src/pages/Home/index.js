import React from "react";
import styled from "styled-components";
import ReactRouterPropTypes from "react-router-prop-types";
import logo from "../../assets/xy-logo.png";
import poster from "../../assets/xy-bg.png";

export default function Home({ history }) {
  const handleClick = () => {
    history.push("/lobby");
  };

  return (
    <Wrapper>
      <GameIntro>
        <div>
          <h1>
            <Logo src={logo} alt="logo" />
          </h1>
          <Content>
            <p className="game-intro">Win as much as you can</p>
            <Poster src={poster} alt="game-poster" />
          </Content>
        </div>
        <Button type="button" onClick={handleClick}>
          Start
        </Button>
      </GameIntro>
    </Wrapper>
  );
}

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  width: 100%;
  height: 100%;
`;

const GameIntro = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  h1 {
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 2rem;
    margin-bottom: 21px;
  }

  p {
    font-size: 1.5em;
    font-weight: 500;
    text-align: center;
    margin: 1.2em 0 1.5em 0;
    text-transform: uppercase;
  }
`;

const Content = styled.div`
  width: 80%;
  margin: 0 auto;
  img {
    width: 100%;
  }
`;

const Logo = styled.img`
  width: 3em;
  margin-right: 15px;
  border-radius: 10px;
`;

const Button = styled.button`
  width: 25%;
  height: 2.5em;
  font-size: 1.2rem;
  border-radius: 1.2rem;
  background-color: #54628c;
  margin-top: 2em;
  cursor: pointer;
  color: #fff;
  outline: 0;

  &:hover,
  :active {
    color: #f2aeae;
  }
`;

const Poster = styled.img`
  width: 50%;
  height: 50%;
  border-radius: 1.2rem;
`;

Home.propTypes = {
  history: ReactRouterPropTypes.history.isRequired,
};
