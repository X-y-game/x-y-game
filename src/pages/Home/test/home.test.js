import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import Home from "../Home";

describe("Home ", () => {
  it("renders", () => {
    const history = createMemoryHistory();
    const component = renderer.create(<Home history={history} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe("Start button clicked", () => {
    const history = createMemoryHistory();

    beforeEach(() => {
      render(
        <Router history={history}>
          <Home history={history} />
        </Router>
      );
    });

    it(" move to `/channel` route when button is clicked", () => {
      const button = screen.getByText("Start");
      userEvent.click(button);

      expect(history.location.pathname).toBe("/channel");
    });
  });
});
