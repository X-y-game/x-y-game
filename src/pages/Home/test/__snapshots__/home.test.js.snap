// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home  renders 1`] = `
<div
  className="sc-bdvvtL junnHm"
>
  <div
    className="sc-gsDKAQ japwcN"
  >
    <div>
      <h1>
        <img
          alt="logo"
          className="sc-hKwDye cTZbSc"
          src="xy-logo.png"
        />
      </h1>
      <div
        className="sc-dkPtRN lnpABA"
      >
        <p
          className="game-intro"
        >
          Win as much as you can
        </p>
        <img
          alt="game-poster"
          className="sc-jRQBWg fzdYDm"
          src="xy-bg.png"
        />
      </div>
    </div>
    <button
      className="sc-eCImPb gofXFp"
      onClick={[Function]}
      type="button"
    >
      Start
    </button>
  </div>
</div>
`;
